src/backend/distributed/utils/resource_lock.c

```diff
@@ -188,15 +221,12 @@ lock_shard_resources(PG_FUNCTION_ARGS)
 		int64 shardId = DatumGetInt64(shardIdArrayDatum[shardIdIndex]);
 
 		/*
-		 * We don't want random users to block writes. The callers of this
-		 * function either operates on all the colocated placements, such
-		 * as shard moves, or requires superuser such as adding node.
-		 * In other words, the coordinator initiated operations has already
-		 * ensured table owner, we are preventing any malicious attempt to
-		 * use this function.
+		 * We don't want random users to block writes. If the current user
+		 * has privileges to modify the shard, then the user can already
+		 * acquire the lock. So, we allow.
 		 */
 		bool missingOk = true;
-		EnsureShardOwner(shardId, missingOk);
+		EnsureCurrenUserCanModifyShard(shardId, missingOk);
```

Comment:
We solved this in a different way: https://github.com/citusdata/citus/pull/6293/files#diff-12c9341f39a8ded1ca3cc8ab1ee87a4cd710f58989851fe29d31790954cdeed4R253